src/algebra/algebra/operations.lean

```diff
@@ -289,6 +332,43 @@ protected theorem pow_induction_on
 submodule.pow_induction_on' M
   (by exact hr) (λ x y i hx hy, hadd x y) (λ m hm i x hx, hmul _ hm _) hx
 
+/-- `submonoid.map` as a `monoid_hom`, when applied to `alg_hom`s. -/
+@[simps]
+def map_monoid_hom {A'} [semiring A'] [algebra R A'] (f : A →ₐ[R] A') :
```

Comment:
You should be able to make this a `monoid_with_zero_hom` since map also preserves 0 (` = ⊥`).

Reply:
Good idea, done.